### PR TITLE
Core: replace stray cout in config

### PR DESCRIPTION
**Pull request by selsta**
`[2017-11-04 01:22:00] <anonimal> selsta: it was probably just missed, I don't know`

---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the contributor guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

File: src/core/util/config.cc [hunk 1]
```diff
@@ -161,7 +161,9 @@ void Configuration::ParseConfig()
   // Help options
   if (m_Map.count("help"))
     {
-      std::cout << config_options << std::endl;
+      // Logging is not setup yet
```
Comment (anonimal) [inline src/core/util/config.cc]: Comments should describe what's being done or will be done, not what's *not* being done.

File: src/core/util/config.cc [hunk 2]
```diff
@@ -161,7 +161,9 @@ void Configuration::ParseConfig()
...
+      core::SetupLogging(m_Map);
```
Comment (anonimal) [inline src/core/util/config.cc]: Another option is to move logging initialization from core instance initialization into configuration initialization (e.g., in this function right after the map is notified). This way we only need to call the function once. An argument against doing so would be that the logging setup is then not decoupled from config parsing - but initialization requires a parsed map and per-instance configurations would require per-logging configurations anyway (eventually, when we're not using the global logging singleton).

File: src/core/util/config.cc (hunk 1 above)
Comment (selsta) [reply src/core/util/config.cc]: I always write comments to why I’m doing things.. what’s being done can be read from the code anyway?

File: src/core/util/config.cc (hunk 2 above)
Comment (selsta) [reply src/core/util/config.cc]: Changed it as you suggested.

File: src/core/instance.cc [hunk 3]
```diff
@@ -56,10 +56,6 @@ Instance::Instance(const std::vector<std::string>& args) try
     : m_Exception(__func__),
       m_Config(args)
   {
-    // TODO(anonimal): Initializing of sources/streams/sinks must come after we've properly configured the logger.
-    //   we do this here so we can catch debug logging before instance "initialization". This is not ideal
```
Comment (anonimal) [inline src/core/instance.cc]: Please don't remove this TODO. It is not yet resolved. You can move it along with the function.

File: src/core/util/config.cc (hunk 1 above)
Comment (anonimal) [reply src/core/util/config.cc]: >I always write comments to why I’m doing things

Well it's good to get into better habits then. And if you're going to describe why, then describe why. Obviously "logging is not setup yet" but that doesn't explain *why* you need to setup logging.

Comment (selsta) [reply src/core/util/config.cc]: Yea, that comment could have been better.

**Comment by selsta** [comment]
sorry, fixed both things

File: src/core/util/config.cc [hunk 4]
```diff
@@ -158,10 +158,14 @@ void Configuration::ParseConfig()
   bpo::store(
       bpo::command_line_parser(m_Args).options(cli_options).run(), m_Map);
   bpo::notify(m_Map);
+
+  // TODO(anonimal): Initializing of sources/streams/sinks must come after we've properly configured the logger.
+  //   we do this here so we can catch debug logging before instance "initialization". This is not ideal
+  core::SetupLogging(m_Map);
```
Comment (anonimal) [inline src/core/util/config.cc]: Oops! Slight problem: now log options won't be set by config file. My fault! I should've looked before commenting.

We have two options now:
1. Setup logging after we parse config file *then* run help. This is a waste of cycles when running help.
2. Revert all this work and keep your original PR (minus the comment and call to SetupLogging). If we don't initialize custom logging, boost.log will use perfectly fine defaults. So, unless you want a colorful help section, just use boost.log defaults by calling LOG without SetupLogging.

Does that make sense?

Comment (selsta) [reply src/core/util/config.cc]: I changed it back to the 2. option.

**Pull request merged**